### PR TITLE
Cancancan

### PR DESCRIPTION
**Pull request by hitshootnum1**

### CODE REVIEW

**Comment by QuangVinh93** [comment]
ok

**Comment by HUSTToanLH** [comment]
ok

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
```
Comment (dieunb) [inline app/models/ability.rb]: remove `()`

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    user ||= User.new
+    if user.role? :supervisor
```
Comment (dieunb) [inline app/models/ability.rb]: `user.supervisor?`

File: config/database.yml.example [hunk 3]
```diff
@@ -1,25 +0,0 @@
-# SQLite version 3.x
```
Comment (dieunb) [inline config/database.yml.example]: don't remove this file

File: app/models/user.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 class User < ActiveRecord::Base
 
+  ROLES = %w[trainee supervisor]
```
Comment (dieunb) [inline app/models/user.rb]: read rails enum and replace it with this

**Pull request merged**